source/DamageProfile.h

```diff
@@ -0,0 +1,69 @@
+/* DamageProfile.h
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef DAMAGE_PROFILE_H_
+#define DAMAGE_PROFILE_H_
+
+#include "Point.h"
+#include "Projectile.h"
+#include "Weather.h"
+
+class DamageDealt;
+class Ship;
+class Weapon;
+
+
+
+// A class that calculates how much damage a ship should take given the ship's
+// attributes and the weapon it was hit by for each damage type. Bundles the
+// results of these calculations into a DamageDealt object.
+class DamageProfile {
+public:
+	// Constructor for damage taken from a weapon projectile.
+	DamageProfile(const Projectile::ImpactInfo &info);
+	// Constructor for damage taken from a hazard.
+	DamageProfile(const Weather::ImpactInfo &info);
```

Comment:
```suggestion
	DamageProfile(Projectile::ImpactInfo info);
	// Constructor for damage taken from a hazard.
	DamageProfile(Weather::ImpactInfo info);
```
These constructors are called with rvalues so we can take them by value without copies. Then you can also `std::move` the Point fields, as you had previously.